### ISSUE TITLE
Page title and content interface languages are still hardcoded

### ISSUE THREAD

**Issue by Woseseltops** [Signbank/Global-signbank#702]
I have always assumed that these were linked to the modeltranslation package, but upon closer inspection it appears they are not! @vanlummelhuizen and @susanodd , can you verify?

**Comment by Woseseltops** [comment]
Similar to #658 (although I hope the solution will be less drastic), crucial for #696

**Comment by vanlummelhuizen** [comment]
> I have always assumed that these were linked to the modeltranslation package, but upon closer inspection it appears they are not! @vanlummelhuizen and @susanodd , can you verify?

@Woseseltops I can confirm. A little more detail: title and content are fields of the `Page` model and they have translations in fields `title_dutch`, `title_chinese` and `content_dutch` and `content_chinese` in the same model. So, Page is a excellent candidate to move to *Django-modeltranslation*. Since I am already doing this for `FieldChoice` and the like, I can do this for `Page` as well. Perhaps I will do this is a separate branch from the work for #658 and #661 so that it is done before those issues are finished, because they still need a lot of work.

**Comment by Woseseltops** [comment]
>  I can do this for Page as well. Perhaps I will do this is a separate branch from the work for #658 and #661 so that it is done before those issues are finished, because they still need a lot of work.

Sounds very good!

**Comment by vanlummelhuizen** [comment]
I just realized that for #658 and #661 I have quite some migrations. Do something similar for `Page` in a separate branch, would result in clashing migrations. I am going to think about this some more...

**Comment by susanodd** [comment]
Yes, my branch with colors has diverged quite a bit. I stopped pushing it because I keep revising the code.

**Comment by vanlummelhuizen** [comment]
As I see it I could do either of two things:

1. Do this as an extension of the work in #658 and #661. It will take a while before it will be done.
2. I will do it in a separate branch (from master). This means I don't have to wait for #658 and #661 to be finished. Downsides: merging migration from parallel branches could be difficult (although I think is will be ok since they are dealing with different things), and, because of the work on this issue, #658 and #661 will be finished later.

@Woseseltops thoughts?

**Comment by ocrasborn** [comment]
@vanlummelhuizen , option 1 is fine with me, if OK for @Woseseltops

**Comment by susanodd** [comment]
The model translations are all active and these are visible in the Admin. That's where the page titles and field choice names are and can be modified.

Once upon a time @Woseseltops had made a command to export to csv the "field choice names" as found in the database (as can be edited in Admin).
That command was removed during reworking of the model translations since it had hard-coded fields and seemed obsolete.

**Comment by susanodd** [comment]
The only place now where some hard-coded language mentions exist is in the menu bar.

I recently modified the names of the two menu bar search fields there.
To my surprise, the "form" that includes them isn't referenced directly in any other code.
But the get_queryset method of GlossListView inspects them as part of GET.
It seems a bit that these have been evolving. At one point, the search term entered that was also put into the separate language fields.
The implementation of the search (how it is chosen to be implemented), these two search fields are multilingual (not specific to a particular language, as opposed to purposefully being implemented that way).

https://github.com/Signbank/Global-signbank/blob/38f11838ff667c3c6333d9e8fcc3fc6d678256ee/templates/global-templates/menu.html#L79-L85

Probably that can just be replace with just the else clause.

**Comment by susanodd** [comment]
Oops. No, the page fields are language-specific. The page model was not part of the FieldChoice model translation extravaganza to get rid of hard-coded languages.

This has the side effect that the above ought to check the two new interface languages rather than default to English.

**Comment by vanlummelhuizen** [comment]
@susanodd Let me work on this on a separate branch.

**Comment by susanodd** [comment]
This is live now!!

Very elegant, @vanlummelhuizen !!
